### ISSUE TITLE
Support calculating approximate duration

### ISSUE THREAD

**Issue by mycoboco** [Borewit/music-metadata#35]
Giving the `duration: true` option makes all media files be fully scanned to calculate their accurate duration, but when a large number of files should be handled, it is enough to get approximate duration of songs; most clients to play them do not trust what the server gives anyway.

How about adding an option like `approximate: true` to `duration: true` for approximate duration? As I recall, an old version of `musicmetadata` did that by assuming CBR when the first three frames have the same bit-rate; I ported that to `mp3-duration` (https://github.com/mycoboco/mp3-duration/commit/abdb1bca7d9739dbff1b99479b84c811d2bc6f12).

**Comment by Borewit** [comment]
It should already work according your suggestion.

Taken from [MpegParser.ts](https://github.com/Borewit/music-metadata/blob/master/src/mpeg/MpegParser.ts):421

````TypeScript
if (this.frameCount === 3) {
        // the stream is CBR if the first 3 frame bitrates are the same
        if (this.areAllSame(this.bitrates)) {
          // subtract non audio stream data from duration calculation
          // The 128 bytes is based on a very unsafe assumtion that a ID3v1 will follow (ToDo: improve)
          const size = this.tokenizer.fileSize - (this.headerSize === 0 ? 128 : this.headerSize);
          this.format.duration = (size * 8) / header.bitrate;
          this.format.codecProfile = "CBR";
          return; // Done
        } else if (!this.readDuration) {
          return; // Done
        }
      }
````
I will ensure that clear test cases will demonstrate this, that does not seem to be the case.

**Comment by mycoboco** [comment]
Oh, I thought that approximation for CBR got removed because `music-metadata` took much more time to scan 4000+ mp3 files than its old version did; in fact, I couldn't see the scanning finish due to the "out-of-memory" error. Probably, this issue is related to #34. I'll check again after #34 resolved.

**Comment by Borewit** [comment]
That could very well be the reason.

On a good quality (using a recent Lame encoder) MP3 encoded file, it will do the duration calculation on the VBR/Xing/Info header. That header specifies the number of frames.

I just removed a silly rounding from that calculation. It is pretty accurate only thing what is missing is subtracting the "empty" space in the very last frame.

**Comment by mycoboco** [comment]
The test result with the latest release, `0.8.2` is very impressive!

To fully scan 4500+ files,
- `music-metadata` + `mp3-duration` for fallback takes only 7mins
- while `musicmetadata` + `mp3-duration` for fallback took 20mins.

I think I can get rid of `mp3-duration` from `canary` even if `music-metadata` still fails to calculate playtime for 6 files; I'll open this as a separate issue soon.

Thanks for the outstanding work!

**Comment by Borewit** [comment]
@mycoboco, thanks for your reply and promising the benchmark.

Feel free to open an issue for the remaining 6 files or re-open this one.
